src/kv/idb-store.ts

```diff
@@ -0,0 +1,155 @@
+import {deleteSentinel, WriteImplBase} from './write-impl-base.js';
+import type {Read, Store, Write} from './store.js';
+
+const RELAXED = {durability: 'relaxed'};
+const OBJECT_STORE = 'chunks';
+
+const enum WriteState {
+  OPEN,
+  COMMITTED,
+  ABORTED,
+}
+
+export class IDBStore implements Store {
+  private readonly _db: Promise<IDBDatabase>;
+
+  constructor(name: string) {
+    this._db = openDatabase(name);
+  }
+
+  async read(): Promise<Read> {
+    const db = await this._db;
+    const tx = db.transaction(OBJECT_STORE, 'readonly');
+    return new ReadImpl(tx);
+  }
+
+  async write(): Promise<Write> {
+    const db = await this._db;
+    // TS does not have type defs for the third options argument yet.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore Expected 1-2 arguments, but got 3.ts(2554)
+    const tx = db.transaction(OBJECT_STORE, 'readwrite', RELAXED);
+    return new WriteImpl(tx);
+  }
+
+  async close(): Promise<void> {
+    (await this._db).close();
+  }
+}
+
+class ReadImpl {
+  private readonly _tx: IDBTransaction;
+
+  constructor(tx: IDBTransaction) {
+    this._tx = tx;
+  }
+
+  async has(key: string): Promise<boolean> {
+    return (await wrap(objectStore(this._tx).count(key))) > 0;
+  }
+
+  async get(key: string): Promise<Uint8Array | undefined> {
+    return wrap(objectStore(this._tx).get(key));
+  }
+
+  release(): void {
+    // Do nothing. We rely on IDB locking.
```

Comment:
Is there a way to test whether `tx` is open and throw if closed? That would probably be useful for debugging.

Reply:
I looked into this... There are many code paths in Rust were we neither call commit or rollback. The reason this is not a problem is because rollback is not really needed. IDB auto closes the transaction and memstore just drops the pending map when the Write is dropped.

It could become an issue in the future so maybe I should fix all the Rust code paths?

Reply:
I think actually the other way, if there's no real need for an explicit rollback, let's remove it. That's what I was getting at with my other question about the API design. What if we just have `release()` and `commit()`. I don't think there's a need for an explicit rollback.

Sorry I realize this PR has gone on for awhile. Mostly because I was on break.

Reply:
Looking at Rust now.

Reply:
> I don't think there's a need for an explicit rollback.

release is sync and rollback is async.

We cannot make release async.